UnitTests/OIDTVAuthorizationRequestTests.m

```diff
@@ -0,0 +1,198 @@
+
+// Ignore warnings about "Use of GNU statement expression extension" which is raised by our use of
+// the XCTAssert___ macros.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wgnu"
+
+/*! @brief Test value for the @c authorizationEndpoint property.
+ */
+static NSString *const kInitializerTestTVAuthEndpoint = @"https://www.example.com/device/code";
+
+/*! @brief Test value for the @c tokenEndpoint property.
+ */
+static NSString *const kInitializerTestTokenEndpoint = @"https://www.example.com/token";
+
+/*! @brief Test key for the @c additionalParameters property.
+ */
+static NSString *const kTestAdditionalParameterKey = @"A";
+
+/*! @brief Test value for the @c additionalParameters property.
+ */
+static NSString *const kTestAdditionalParameterValue = @"1";
+
+/*! @brief Test value for the @c clientID property.
+ */
+static NSString *const kTestClientID = @"ClientID";
+
+/*! @brief Test value for the @c clientID property.
+ */
+static NSString *const kTestClientSecret = @"ClientSecret";
+
+/*! @brief Test value for the @c scope property.
+ */
+static NSString *const kTestScope = @"Scope";
+
+/*! @brief Test value for the @c scope property.
+ */
+static NSString *const kTestScopeA = @"ScopeA";
+
+/*! @brief Expected HTTP Method for the authorization URLRequest
+ */
+static NSString *const kHTTPPost = @"POST";
+
+/*! @brief Expected ContentType header key for the authorization URLRequest
+ */
+static NSString *const kHTTPContentTypeHeaderKey = @"Content-Type";
+
+/*! @brief Expected ContentType header key for the authorization URLRequest
+ */
+static NSString *const kHTTPContentTypeHeaderValue =
+    @"application/x-www-form-urlencoded; charset=UTF-8";
+
+@implementation OIDTVAuthorizationRequestTests
+
+- (OIDTVServiceConfiguration *)testServiceConfiguration {
+  NSURL *tokenEndpoint = [NSURL URLWithString:kInitializerTestTokenEndpoint];
+  NSURL *TVAuthorizationEndpoint = [NSURL URLWithString:kInitializerTestTVAuthEndpoint];
+
+  OIDTVServiceConfiguration *configuration =
+      [[OIDTVServiceConfiguration alloc] initWithTVAuthorizationEndpoint:TVAuthorizationEndpoint
+                                                           tokenEndpoint:tokenEndpoint];
+  return configuration;
+}
+
+- (NSDictionary<NSString *, NSString *> *)bodyParametersFromURLRequest:(NSURLRequest *)urlRequest {
+  NSString *bodyString = [[NSString alloc] initWithData:urlRequest.HTTPBody
+                                               encoding:NSUTF8StringEncoding];
+  NSArray *bodyParameterStrings = [bodyString componentsSeparatedByString:@"&"];
+  
+
+  NSMutableDictionary<NSString *, NSString *> *bodyParameters = [[NSMutableDictionary alloc] init];
+
+  for (NSString *paramString in bodyParameterStrings) {
+    NSArray *components = [paramString componentsSeparatedByString:@"="];
+    NSLog(@"%@:%@", components[0], components[1]);
+    
+    [bodyParameters setObject:components[1] forKey:components[0]];
```

Comment:
nit: setObject:forKey is an older style. I think it's cleaner to use bodyParameters[components[0] = components[1].